src/main/java/uk/gov/justice/laa/crime/cfecrime/utils/RequestHandler.java

```diff
@@ -2,44 +2,98 @@
-            result.setOutcome(oc);
-            response.withResult(result);
         }
-        return response;
+        return cfeCrimeResponse;
     }
 
-    private static Outcome getOutcomeFromAgeAndPassportedBenefit(Boolean clientUnder18, Boolean clientPassportedBenefit){
+    private static Outcome getOutcomeFromAgeAndPassportedBenefit(Boolean clientUnder18, Boolean clientPassportedBenefit) {
         Outcome outcome = null;
-        if (clientUnder18 != null && clientUnder18.booleanValue()){
-            outcome = Result.Outcome.ELIGIBLE;
+        if (clientUnder18 != null && clientUnder18.booleanValue()) {
+            outcome = Outcome.ELIGIBLE;
         }
-        if (clientPassportedBenefit !=null && clientPassportedBenefit.booleanValue()){
-            outcome = Result.Outcome.ELIGIBLE;
+        if (clientPassportedBenefit != null && clientPassportedBenefit.booleanValue()) {
+            outcome = Outcome.ELIGIBLE;
         }
         return outcome;
     }
+
+    public static void setFullMeansTestOutcome(CfeCrimeRequest cfeCrimeRequest, CfeCrimeResponse cfeCrimeResponse) {
+
+        Outcome fullOutcome = null;
+
+        if (cfeCrimeRequest.getSectionFullMeansTest() != null){
+            FullAssessmentResult statelessInitialResult = statelessApiResponse.getFullMeansAssessment().getResult();
+            fullOutcome = FullMeansTestOutcomeCalculator.getFullMeansTestOutcome(statelessInitialResult, caseType, magCourtOutcome);
+            cfeCrimeResponse.setOutcome(fullOutcome);
+        }
+        cfeCrimeResponse.setOutcome(fullOutcome);
+
+    }
+
+    public static void setInitialMeansTestOutcome(CfeCrimeRequest cfeCrimeRequest, CfeCrimeResponse cfeCrimeResponse) throws UndefinedOutcomeException {
+
+        Outcome initOutcome = null;
+        if (cfeCrimeRequest.getSectionInitialMeansTest() != null) {
+            caseType = CaseType.valueOf(cfeCrimeRequest.getSectionInitialMeansTest().getCaseType().name());
+            if (cfeCrimeRequest.getSectionInitialMeansTest().getMagistrateCourtOutcome() != null) {
+                magCourtOutcome = MagCourtOutcome.valueOf(cfeCrimeRequest.getSectionInitialMeansTest().getMagistrateCourtOutcome().name());
+            }
+            InitAssessmentResult initAssessmentResult = statelessApiResponse.getInitialMeansAssessment().getResult();
+            initOutcome = InitMeansTestOutcomeCalculator.getInitMeansTestOutcome(initAssessmentResult, statelessApiResponse.getInitialMeansAssessment().isFullAssessmentPossible());
+        }
+        cfeCrimeResponse.setOutcome(initOutcome);
+
+    }
+
+    public static void setCmaResponse(boolean fullAssessmentPossible, FullAssessmentResult fullAssessmentResult, InitAssessmentResult initAssessmentResult){
+        cmaService.setFullAssessmentPossible(fullAssessmentPossible);
+        cmaService.setFullAssessmentResult(fullAssessmentResult);
+        cmaService.setInitAssessmentResult(initAssessmentResult);
+    }
```

Comment:
don't think these methods belong on requestHandler - I think they belong on StubCMA, and the CMA interface should then be a parameter to handleRequest to allow the caller to pass stub/real CMA